modules/javafx.controls/src/main/java/javafx/scene/control/Skin.java

```diff
@@ -60,6 +73,14 @@
      */
     public Node getNode();
 
+    /**
+     * Called by {@link Control#setSkin(Skin)} on a pristine control, or after the
+     * previous skin has been uninstalled via its {@link #dispose()} method.
```

Comment:
Hmm .. as formulated that sounds to me like an implementation detail of Control (which shouldn't appear here). We might think of something unrelated to specific users.

Also: the congenial partner of a Skin is a Skinnable - all narrowing to a special implementator of Skinnable here should be avoided. We have an open issue for the invers, that is Skinnable overspecifying itself as Control [JDK-8260364](https://bugs.openjdk.org/browse/JDK-8260364)

Reply:
Reworded the description to refer to Skinnable.setSkin().  I feel it is important to mention where in the life cycle of the Skin the install() is getting invoked, without changing too much, but any suggestions will be greatly appreciated!